examples/o1/Using_chained_calls_for_o1_structured_outputs.ipynb

```diff
@@ -0,0 +1,266 @@
+     ]
+    }
+   ],
+   "source": [
+    "import requests\n",
+    "from pydantic import BaseModel\n",
+    "from openai import OpenAI\n",
+    "\n",
+    "client = OpenAI()\n",
+    "\n",
+    "def fetch_html(url):\n",
+    "    response = requests.get(url)\n",
+    "    if response.status_code == 200:\n",
+    "        return response.text\n",
+    "    else:\n",
+    "        return None\n",
+    "\n",
+    "url = \"https://en.wikipedia.org/wiki/List_of_largest_companies_in_the_United_States_by_revenue\"\n",
+    "html_content = fetch_html(url)\n",
+    "\n",
+    "class CompanyData(BaseModel):\n",
+    "    company_name: str\n",
+    "    page_link: str\n",
+    "    reason: str\n",
+    "\n",
+    "class CompaniesData(BaseModel):\n",
+    "    companies: list[CompanyData]\n",
+    "\n",
+    "json_format = \"\"\"\n",
+    "{\n",
+    "    \\\"company_name\\\": \\\"OpenAI\\\",\n",
+    "    \\\"page_link\\\": \\\"https://en.wikipedia.org/wiki/OpenAI\\\",\n",
+    "    \\\"reason\\\": \\\"OpenAI would benefit because they are an AI company...\\\"}\"\n",
+    "\"\"\"\n",
+    "\n",
+    "o1_response = client.chat.completions.create(\n",
+    "    model=\"o1-preview\",\n",
+    "    messages=[\n",
+    "        {\n",
+    "            \"role\": \"user\", \n",
+    "            \"content\": f\"\"\"\n",
+    "You are a business analyst designed to understand how AI technology could be used across large corporations.\n",
+    "\n",
+    "Read the following html and return which companies would benefit from using AI technology: {html_content}. Rank these propects by opportunity by comparing them and show me the top 3. Return each company with the following information: {CompanyData.model_fields.keys()}\"\n",
+    "\"\"\"\n",
+    "        }\n",
+    "    ]\n",
+    ")\n",
+    "\n",
+    "o1_response_content = o1_response.choices[0].message.content\n",
+    "\n",
+    "response = client.beta.chat.completions.parse(\n",
+    "    model=\"gpt-4o-2024-08-06\",\n",
+    "    messages=[\n",
+    "        {\n",
+    "            \"role\": \"user\", \n",
+    "            \"content\": f\"\"\"\n",
+    "Given the following data, format it with the given response format: {o1_response_content}\n",
+    "\"\"\"\n",
+    "        }\n",
+    "    ],\n",
+    "    response_format=CompaniesData,\n",
+    ")\n",
+    "\n",
+    "print(response.choices[0].message.parsed)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Conclusion\n",
+    "\n",
+    "The o1 models don't currently have structured outputs support, but we can use the existing structured outputs functionality from `gpt-4o`/`gpt-4o-mini` by chaining two requests together. \n",
```

Comment:
I would add either here or in the intro a blurb on what the importance of schema adherence is. Why do users want/need it, when it's beneficial, etc. This can help frame the problem for people reading.

Reply:
Sounds good!